### PR TITLE
Modify ExportCmd to export genesis with bech32_prefix

### PR DESCRIPTION
**Pull request by anhductn2001**
# PR Standards
Since Exported type of Cosmos-SDK default return format is GenesisDoc which does not include the `bech32_prefix` field, we need to modify it to allow `bech32_prefix` to be exported every time we export genesis

## Opening a pull request should be able to meet the following requirements

---

Closes #XXX

**All** items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow-up issues.

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

**Comment by danwt** [comment]
Is there an issue?

File: server/commands.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 	"github.com/cosmos/cosmos-sdk/server"
 	"github.com/cosmos/cosmos-sdk/server/types"
 	"github.com/cosmos/cosmos-sdk/version"
+	rdk_types "github.com/dymensionxyz/dymension-rdk/server/types"
```
Comment (danwt) [inline server/commands.go:11]: rdktypes

File: server/commands.go [hunk 2]
```diff
@@ -19,7 +20,7 @@ import (
 )
 
 // add Rollapp commands
```
Comment (danwt) [inline server/commands.go:22]: wrong docstr

**Comment by anhductn2001** [comment]
yea, it relate to: https://github.com/dymensionxyz/rollapp-evm/issues/178

File: server/export.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+// DONTCOVER
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+
+	rdk_genutiltypes "github.com/dymensionxyz/dymension-rdk/x/genutil/types"
```
Comment (danwt) [inline server/export.go:11]: no snake

File: server/commands.go [hunk 4]
```diff
@@ -19,7 +20,7 @@ import (
...
-func AddRollappCommands(rootCmd *cobra.Command, defaultNodeHome string, appCreator types.AppCreator, appExport types.AppExporter, addStartFlags types.ModuleInitFlags) {
+func AddRollappCommands(rootCmd *cobra.Command, defaultNodeHome string, appCreator types.AppCreator, appExport rdk_types.AppExporter, addStartFlags types.ModuleInitFlags) {
 	dymintCmd := &cobra.Command{
```
Comment (danwt) [inline server/commands.go:24]: does this have usages?

File: server/export.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+					return err
+				}
+
+				fmt.Println(string(genesis))
+				return nil
+			}
+
+			genDoc, err := rdk_genutiltypes.GenesisDocFromFile(config.GenesisFile())
+			if err != nil {
+				fmt.Println("Failed to read genesis doc from file", err)
```
Comment (danwt) [inline server/export.go:65]: why sometimes stderr, sometimes stdout?

File: server/export.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+				return err
+			}
+
+			height, _ := cmd.Flags().GetInt64(FlagHeight)
+			forZeroHeight, _ := cmd.Flags().GetBool(FlagForZeroHeight)
+			jailAllowedAddrs, _ := cmd.Flags().GetStringSlice(FlagJailAllowedAddrs)
+
+			exported, err := appExporter(serverCtx.Logger, db, traceWriter, height, forZeroHeight, jailAllowedAddrs, serverCtx.Viper)
+			if err != nil {
+				return fmt.Errorf("error exporting state: %v", err)
```
Comment (danwt) [inline server/export.go:80]:
```suggestion
				return fmt.Errorf("exporting state: %v", err)
```

Comment (danwt) [inline server/export.go:80]:
```suggestion
				return fmt.Errorf("exporting state: %w", err)
```

File: server/export.go [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+				return err
+			}
+
+			doc["app_state"] = exported.AppState
+			doc["validators"] = exported.Validators
+			doc["initial_height"] = exported.Height
+
+			consensus_param := doc["consensus_params"].(map[string]interface{})
+			block := consensus_param["block"].(map[string]interface{})
+			time_iota_ms, _ := strconv.ParseInt(block["time_iota_ms"].(string), 10, 64)
```
Comment (danwt) [inline server/export.go]: no snake

Need to check OK for type assertion?

File: server/types/app.go [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+type (
+	// AppOptions defines an interface that is passed into an application
+	// constructor, typically used to set BaseApp options that are either supplied
+	// via config file or through CLI arguments/flags. The underlying implementation
+	// is defined by the server package and is typically implemented via a Viper
+	// literal defined on the server Context. Note, casting Get calls may not yield
+	// the expected types and could result in type assertion errors. It is recommend
+	// to either use the cast package or perform manual conversion for safety.
+	AppOptions interface {
+		Get(string) interface{}
```
Comment (danwt) [inline server/types/app.go:22]:
```suggestion
		Get(string) any
```

File: x/genutil/types/genesis_state.go [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+	"fmt"
+
+	tmos "github.com/tendermint/tendermint/libs/os"
+)
+
+// GenesisStateFromGenFile creates the core parameters for genesis initialization
+// for the application.
+//
+// NOTE: The pubkey input is this machines pubkey.
+func GenesisStateFromGenFile(genFile string) (genesisState map[string]json.RawMessage, genDoc map[string]interface{}, err error) {
```
Comment (danwt) [inline x/genutil/types/genesis_state.go:14]: if you do concrete type assertions without checking `ok` on the genesisState, can we have a concrete type instead?

**Review by danwt — COMMENTED** [review]
I'm kind of missing the TLDR here
what is the problem being solved?

File: server/commands.go (hunk 2 above)
Comment (anhductn2001) [reply server/commands.go:22]: It was a previous comment, but I'll add some comments to it

File: server/commands.go (hunk 4 above)
Comment (anhductn2001) [reply server/commands.go:24]: Yes, see it: https://github.com/dymensionxyz/rollapp-evm/pull/188/files#diff-3ef291e710a554004d50d61607fe864a27e6b4350f3cb764f2cff8b442c2c213L215

File: server/export.go (hunk 5 above)
Comment (anhductn2001) [reply server/export.go:65]: it's just a function taken from cosmos-sdk with a little change

File: server/export.go (hunk 6 above)
Comment (anhductn2001) [reply server/export.go:80]: I think we shouldn't change it

**Comment by anhductn2001** [comment]
I have updated description of this PR.

**Pull request closed without merging**